### DISCUSSION TITLE
Compile fail

### DISCUSSION THREAD

**Discussion by Goldi-2022** [PlummersSoftwareLLC/NightDriverStrip#225] [General]
I am new to this, so please forgive me if this is the wrong place to look for help.

- I have a M5Stick-plus wired up as per Dave P's Laser Cannon video.
- I've cloned the project using GIT Desktop (today).
- Changed the PlatformIO.ini default environment to m5plusdemo. 

When I try to compile the code I get these error messages, as the compile fail ...

*** [.pio\build\m5plusdemo\src\effects.cpp.o] Error 1
cc1plus.exe: some warnings being treated as errors
*** [.pio\build\m5plusdemo\src\ledmatrixgfx.cpp.o] Error 1
cc1plus.exe: some warnings being treated as errors
*** [.pio\build\m5plusdemo\src\colordata.cpp.o] Error 1
cc1plus.exe: some warnings being treated as errors
*** [.pio\build\m5plusdemo\src\audio.cpp.o] Error 1
cc1plus.exe: some warnings being treated as errors
*** [.pio\build\m5plusdemo\src\drawing.cpp.o] Error 1
cc1plus.exe: some warnings being treated as errors
*** [.pio\build\m5plusdemo\src\main.cpp.o] Error 1

If I change nothing and compile again, I get a different set of error messages.

I hope someone will have the knowledge and patience to help me.

**Comment by davepl** [comment]
Can you try building the spectrum project as a test?  And please include the actual compiler errors, if possible, that’d make it easier to sort out!

- Dave

**Reply by Goldi-2022** [reply]
Thank you very much for taking the time to help Dave.

I've changed the environment to spectrum and recompiled.  The enclosed text file contains the full compile output from VSCode.

Thanks again, Derek

[Compile Text.txt](https://github.com/PlummersSoftwareLLC/NightDriverStrip/files/10791677/Compile.Text.txt)

**Comment by davepl** [comment]
The source tree itself compiles fine, so there’s something about the setup on your machine that must be wonky, I just can’t guess as to what from the errors!

The two things I would check:

- First, make sure you have free space on your source drive, ideally several gig

Run:
pio upgrade

- Make sure you wind up with PlatformIO 6.1.6 or later, then:

Run:
pio pkg update

Run:
pio run -t clean

Then try building again, let me know if it changes!

**Reply by Goldi-2022** [reply]
Thank you very much Dave.

Having done what you suggested, I now have an operating spectrum analyser displaying on the M5stick-plus.

I really appreciate you taking the time to help me.

Thanks again, Derek.
